routes/authentication.rb

```diff
@@ -0,0 +1,49 @@
+class App < Sinatra::Base
+  UNAUTHENTICATED_URL       = '/auth/unauthenticated'.freeze
+  EMAIL_UNAVAILABLE_NOTICE  = 'This email is registered.'.freeze
+  INVALID_PARAMETERS_NOTICE = 'Please, fill form properly.'.freeze
+  INVALID_LOGIN_INFO_NOTICE = 'Email or password is incorrect'.freeze
+
+  get '/auth/login' do
+    slim :login
+  end
+
+  post '/auth/login' do
+    env['warden'].authenticate!
+
+    redirect session.fetch(:return_to, '/')
+  end
+
+  get '/auth/logout' do
+    AuthenticationService::Logout.call(env)
+
+    redirect '/'
+  end
+
+  get '/auth/signup' do
+    slim :signup
+  end
+
+  post '/auth/signup' do
+    result = AuthenticationService::SignUp.call(params)
+    
+    case result
+    when :email_unavailable
+      flash[:notice] = EMAIL_UNAVAILABLE_NOTICE
+      redirect '/auth/signup'
```

Comment:
Wrap it into `AfterSignUpHandler` or something similar that returns flash message & redirect path based on the result